pkg/distro/rhel/images.go

```diff
@@ -658,7 +668,15 @@ func ImageInstallerImage(workload workload.Workload,
 		img.AdditionalDrivers = installerConfig.AdditionalDrivers
 	}
 
-	img.AdditionalAnacondaModules = []string{"org.fedoraproject.Anaconda.Modules.Users"}
+	instCust, err := customizations.GetInstaller()
+	if err != nil {
+		return nil, err
+	}
+	if instCust != nil && instCust.Modules != nil {
+		img.AdditionalAnacondaModules = append(img.AdditionalAnacondaModules, instCust.Modules.Enable...)
+		img.DisabledAnacondaModules = append(img.DisabledAnacondaModules, instCust.Modules.Disable...)
+	}
+	img.AdditionalAnacondaModules = append(img.AdditionalAnacondaModules, anaconda.ModuleUsers)
```

Comment:
I'm a bit confused that we add the users module unconditionally for the `ImageInstaller` here; but conditionally for the others (only if kickstart user customizations are enabled). Why is that?

As far as I know one needs to add a user for both image types in Anaconda.

Reply:
The image installer's payload never contains a user account, so we enable it to force interactive user creation during installation, otherwise the user will end up with no way to log in to the machine.  This decision was made to accommodate the IB service which doesn't support user creation (in the UI at least) but does have support for building image installer ISOs.

For other image types (ostree-based installers), the user module is only enabled by default when the blueprint contains a user so that it can be created.  It's not required however to create a user for such payloads because it is possible that a user is already created in the payload itself (the ostree commit).  So if a user has an ostree commit with a user account in it, wants to create an ISO to deploy it, and also wants to make that ISO fully unattended (without creating a user at install time), the user module will have to be disabled for that to be possible.

I'm open to discussing these decisions and reevaluating, especially now that users will have control of the modules that are enabled.

Reply:
I think we can keep it as is as we always build the tar that gets deployed by image installer as well.